File: src/main/java/guru/xgm/language/zinc/lang/ZincCommenter.java
package guru.xgm.language.zinc.lang;

import com.intellij.lang.Commenter;
import org.jetbrains.annotations.Nullable;

public class ZincCommenter implements Commenter {
    @Override
    public @Nullable String getLineCommentPrefix() {
        return "//";
    }

    @Override
    public @Nullable String getBlockCommentPrefix() {
        return "/*";
    }

    @Override
    public @Nullable String getBlockCommentSuffix() {
        return "*/";
    }

    @Override
    public @Nullable String getCommentedBlockCommentPrefix() {
        return "/*";
    }

    @Override
    public @Nullable String getCommentedBlockCommentSuffix() {
        return "*/";
    }
}

File: src/main/kotlin/raft/war/language/zinc/lang/ZincCommenter.kt
package raft.war.language.zinc.lang

import com.intellij.lang.Commenter

class ZincCommenter : Commenter {
    override fun getLineCommentPrefix(): String = "//"
    override fun getBlockCommentPrefix(): String = "/*"
    override fun getBlockCommentSuffix(): String = "*/"
    override fun getCommentedBlockCommentPrefix(): String = "/*"
    override fun getCommentedBlockCommentSuffix(): String = "*/"
}
